viral/ensemble_reactivation.py

```
@@ -0,0 +1,616 @@
+        preactivation = offline_spks_pre[pcs_mask]
+        # ONLINE
+        running_bouts = get_running_bouts(
+            place_cells=place_cells,
+            speed=speed,
+            frames_positions=positions,
+            aligned_trial_frames=aligned_trial_frames,
+            config=config,
+        )
+        ssp_vectors = get_ssp_vectors(place_cells_running=running_bouts)
+        # ensemble_matrix = compute_ICA_components(ssp_vectors=ssp_vectors)
+        ensemble_matrix = compute_PCA_components(ssp_vectors=ssp_vectors)
```

Ruff (0.11.9)

Undefined name `compute_PCA_components`

(F821)